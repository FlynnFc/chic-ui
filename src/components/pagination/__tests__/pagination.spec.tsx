import Pagination from '..';
import { render } from 'test/utils';

describe('Pagination Component', () => {
  it('Renders Correctly', () => {
    const PaginationGroup = render(
      <>
        <div>
          <Pagination
            bgType="success"
            pages={7}
            startCountInZero
            activeBg="white"
            activeColor="black"
            activeHoverColor="black"
            whenNextPage={(page) =>
              alert(`whenNextPage page hit, currently on page ${page}`)
            }
            whenPreviousPage={(page) =>
              alert(`whenPreviousPage page hit, currently on page ${page}`)
            }
            whenPageChange={(page) =>
              alert(`whenPageChange hit, currently on page ${page}`)
            }
          />
        </div>
        <div>
          <Pagination
            customBg={{
              bgColor: 'gray',
              hover: 'black',
              color: 'yellow'
            }}
            pages={4}
            activeBg="white"
            activeColor="black"
            activeHoverColor="black"
            whenNextPage={(page) =>
              alert(`whenNextPage page hit, currently on page ${page}`)
            }
            whenPreviousPage={(page) =>
              alert(`whenPreviousPage page hit, currently on page ${page}`)
            }
            whenPageChange={(page) =>
              alert(`whenPageChange hit, currently on page ${page}`)
            }
          />
        </div>
      </>
    );
    expect(PaginationGroup).toMatchSnapshot();
  });
});
