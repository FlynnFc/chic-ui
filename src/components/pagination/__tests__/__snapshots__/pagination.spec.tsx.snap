// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Renders Correctly 1`] = `
Array [
  <div>
    <div
      className="sc-bdfBQB"
    >
      <button
        className="sc-gsTEea sc-dlfnuX ilamKy bAGomm"
        onClick={[Function]}
      >
        Previous
      </button>
      <button
        className="sc-gsTEea gSoGVl"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="sc-gsTEea ilamKy"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="sc-gsTEea ilamKy"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="sc-gsTEea ilamKy"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="sc-gsTEea ilamKy"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="sc-gsTEea ilamKy"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="sc-gsTEea ilamKy"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="sc-gsTEea sc-hKgJUU ilamKy iXkjoo"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>,
  <div>
    <div
      className="sc-bdfBQB"
    >
      <button
        className="sc-gsTEea sc-dlfnuX fODqpX bAGomm"
        onClick={[Function]}
      >
        Previous
      </button>
      <button
        className="sc-gsTEea gSoGVl"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="sc-gsTEea fODqpX"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="sc-gsTEea fODqpX"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="sc-gsTEea fODqpX"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="sc-gsTEea sc-hKgJUU fODqpX iXkjoo"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>,
]
`;
